### PR TITLE
[LuaCheck]: Various changes and improvements

### PR DESCRIPTION
**Pull request by Panquesito7**
- Add GitHub workflow.
- Fix [LuaCheck](https://github.com/luarocks/luacheck) warnings.
- Added build status badge on `README.md`. [Build status](https://github.com/minetest-mods/digilines/actions)

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-	"dump",
-	"VoxelArea",
+    "vector",
+    "screwdriver",
+    "minetest",
+    "default",
+    "pipeworks",
+    "dump",
+    "VoxelArea",
+    "ItemStack",
```
Comment (SmallJoker) [inline .luacheckrc:10]: Why not tabs?

Comment (Panquesito7) [reply .luacheckrc:10]: AFAIK, I'm using tabs here.

Comment (OgelGames) [reply .luacheckrc:10]: Those are spaces, not tabs, I can select each space individually.

Comment (Panquesito7) [reply .luacheckrc:10]: Probably there's something messed up with my IDE...I am using TABS according to my IDE?

Comment (Panquesito7) [reply .luacheckrc]: Should be fixed now.

**Review by OgelGames — APPROVED** [review]

**Pull request merged**